static/js/latest-news.js

```diff
@@ -0,0 +1,115 @@
+function formatDate(date) {
+  const parsedDate = new Date(date);
+  const monthNames = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December"
+  ];
+
+  const day = parsedDate.getDate();
+  const monthIndex = parsedDate.getMonth();
+  const year = parsedDate.getFullYear();
+
+  return day + " " + monthNames[monthIndex] + " " + year;
+}
+
+function htmlForLatestArticles(articles) {
+  const articlesTree = document.createDocumentFragment();
+  for (let i = 0; i < articles.length; i++) {
+    const article = articles[i];
+    const div = document.createElement("div");
+    div.classList.add("col-3");
+    const header = document.createElement("h4");
```

Comment:
Even with such fairy simple HTML structure it's quite hard to see what the resulting HTML is meant to be when it's manually created like that.

While not ideal, template strings can be used, to it looks much more like HTML, is easier to understand and probably less prone to errors.

This could look something like:
```js
   const header = `<h4><a href="${article.link}">${article.title.rendered}</a></h4>`;
   const div = `<div class="col=3">${ header }${ date }</div>`;
```

At least some of the elements could be built like that before they are turned into DOM and added to document fragment.

Reply:
Seems a lot neater!
Keeps it simple and readable, thanks

Reply:
Updated :)